drivers/bluetooth/hci/Kconfig

```diff
@@ -130,3 +130,16 @@ config BT_RPMSG_NRF53_RX_PRIO
 	default 8
 
 endif # BT_RPMSG_NRF53
+
+config BT_DRIVER_QUIRK_NO_AUTO_DLE
+	bool "Host auto-initiated Data Length Update quirk"
+	depends on BT_AUTO_DATA_LEN_UPDATE
+	default y if BT_RPMSG_NRF53
```

Comment:
Please add && BT_LL_SW_SPLIT

Reply:
BT_LL_SW_SPLIT is not available for CPUAPP. 

Other controller variant used have to set the following in their out-of-tree Kconfig, maybe?
```
config BT_DRIVER_QUIRK_NO_AUTO_DLE
        default n
```

Reply:
I would prefer to have this configuration disabled by default. The host should assume the controller implements the behavior suggested by the specification.

Reply:
This has been already discussed here: https://github.com/zephyrproject-rtos/zephyr/pull/26277

Thinking aloud, Zephyr BT host does not have a public API to set the suggested default data length for new connections, which means for an optimal usecase the Data Length Update Procedure needs to be performed after GATT MTU exchange, else the maximum Tx and Rx Octets will be used between connected devices when using BT_AUTO_DATA_LEN_UPDATE. Using the maximum PDU size implies, under noisy environments radio could be spending more time in Rx due to CRC errors in PDU len field.

Auto initiated data length update in the controller will come in later releases, until then if BT_AUTO_DATA_LEN_UPDATE is enable then BT_DRIVER_QUIRK_NO_AUTO_DLE is required in the upstream Zephyr.

Reply:
The idea was to use bt_conn_le_data_len_update(NULL, &param); to set default params.

@cvinayak This fix is not addressing other HCI split builds, for example nrf91 (h4) would also be affected by it.
Could we instead determine if the quirk should be enabled at runtime, instead of defaulting to default on, and then turning it off for other drivers.
Maybe turn the quirk on for manufacturer ID == 0x05F1?

Reply:
> The idea was to use bt_conn_le_data_len_update(NULL, &param); to set default params.

Correct me, in upstream master the only place Write Suggested Data Length is here in the `le_init`, right? https://github.com/zephyrproject-rtos/zephyr/blob/48e791a625a9ee98d032631ab234d3036696fe3f/subsys/bluetooth/host/hci_core.c#L5634

> quirk should be enabled at runtime

Then it cant be `const` member in hci driver structure, beside that, 

> Maybe turn the quirk on for manufacturer ID == 0x05F1?

manufacturer id can not be used as it can be set to vendor's id.

> instead of defaulting to default on, and then turning it off for other drivers.

This is not new, many upstream board defconfig do similar turning off of features which are default on.
https://github.com/zephyrproject-rtos/zephyr/blob/master/boards/posix/nrf52_bsim/Kconfig.defconfig#L35
https://github.com/zephyrproject-rtos/zephyr/blob/master/soc/xtensa/intel_s1000/Kconfig.defconfig#L31

This way out-of-tree repositories can always set preferred default values.

Reply:
> Then it cant be const member in hci driver structure, beside that,

That's a shame.

But fair enough. Let's go with this. Atleast we have the option to disable it when we want to.